langchain4j/src/test/java/dev/langchain4j/service/AiServicesWithJsonSchemaIT.java

```diff
@@ -71,26 +73,325 @@ void should_extract_pojo_with_primitives() {
+                    person.height.equals(1.78) &&
+                    !person.married
+            );
+
+            assertThat(people).anyMatch(person ->
+                person.name.equals("Franny") &&
+                    person.age == 35 &&
+                    person.height.equals(1.65) &&
+                    person.married
+            );
+
+            verify(model).chat(ChatRequest.builder()
+                .messages(singletonList(userMessage(text)))
+                .responseFormat(ResponseFormat.builder()
+                    .type(JSON)
+                    .jsonSchema(JsonSchema.builder()
+                        .name("Set_of_Person")
+                        .rootElement(JsonObjectSchema.builder()
+                            .addProperty("items", JsonArraySchema.builder()
+                                .items(JsonObjectSchema.builder()
+                                    .addStringProperty("name")
+                                    .addIntegerProperty("age")
+                                    .addNumberProperty("height")
+                                    .addBooleanProperty("married")
+                                    .required("name", "age", "height", "married")
+                                    .build())
+                                .build())
+                            .required("items")
+                            .build())
+                        .build())
+                    .build())
+                .build());
+            verify(model).supportedCapabilities();
+        }
+    }
+
+    interface MaritalStatusExtractor {
+
+        enum MaritalStatus {
+
+            SINGLE, MARRIED
+        }
+
+        List<MaritalStatus> extractMaritalStatusFrom(String text);
+    }
+
+    @Test
+    void should_extract_list_of_enums() {
+
+        for (ChatLanguageModel model : models()) {
+
+            // given
+            model = spy(model);
+
+            MaritalStatusExtractor personExtractor = AiServices.create(MaritalStatusExtractor.class, model);
+
+            String text = "Klaus is 37 years old, 1.78m height and single. " +
+                "Franny is 35 years old, 1.65m height and married." +
+                "Staniel is 33 years old, 1.70m height and married.";
+
+            // when
+            List<MaritalStatusExtractor.MaritalStatus> statuses = personExtractor.extractMaritalStatusFrom(text);
+
+            // then
+            assertThat(statuses).hasSize(3);
+
+            // Use assertThat with filters to ensure each marital status is in the list
+            ArrayList<MaritalStatusExtractor.MaritalStatus> maritalStatuses = new ArrayList<>();
+            maritalStatuses.add(MaritalStatusExtractor.MaritalStatus.SINGLE);
+            maritalStatuses.add(MaritalStatusExtractor.MaritalStatus.MARRIED);
+            maritalStatuses.add(MaritalStatusExtractor.MaritalStatus.MARRIED);
+            assertThat(statuses).containsExactly(MaritalStatusExtractor.MaritalStatus.SINGLE,
```

Comment:
```suggestion
            assertThat(statuses).containsExactly(MaritalStatus.SINGLE,
```
please do static import